lib/cloudregister/registerutils.py

```diff
@@ -741,6 +758,12 @@ def set_proxy():
     return True
 
 
+# ----------------------------------------------------------------------------
+def set_new_registration_flag():
+    """Set a marker that this is the beginning of the registration process"""
+    Path(REGISTRATION_DATA_DIR + NEW_REGISTRATION_MARKER).touch()
```

Comment:
Hmm, but is it actually required?
The call stack before the fix looks something like this:

```
susecloud.RESOLVEURL():
  if utils.get_smt()
    if get_current_smt():
      registered = is_registered(): # <--- False due to has_repos check down below
        if check_registration():
          if has_repos(smt_server_name) and __has_credentials(smt_server_name):
            return True
      
      if registered:
        if is_responsive():
          return current_smt
        else find_equivalent_smt_server():
          return equivalent()
  else utils.get_update_server_name_from_hosts()
    logging.warning('The system is in an inconsistent state')
```

`utils.get_smt` is failing due to nested `has_repos` call, and then it enters the `get_update_server_name_from_hosts` branch that produces the warning.

However, you've replaced `has_repos` with `has_services` -- shouldn't it work properly with just that change?
IMO, having the `NEW_REGISTRATION_MARKER` makes thing more convoluted, it'd be great to avoid extra complexity if possible.